### PR TITLE
Laetitia week 1

### PR DESCRIPTION
**Pull request by supertata**
# Your Details

Your Name:
Your City:
Your Slack Name:

# Homework Details

Module:
Week:

### CODE REVIEW

**Comment by supertata** [comment]
Please ignore "extra exercise"

File: week-1/Homework/extra/1-currency-conversion.js [hunk 1]
```diff
@@ -5,8 +5,11 @@
   Write a function that converts a price to USD (exchange rate is 1.4 $ to £)
 */
 
-function convertToUSD() {}
+function convertToUSD(priceInPound) {
+return (priceInPound * 1.4);
+}
```
Comment (Abdoulrazack95) [inline week-1/Homework/extra/1-currency-conversion.js:10]: Amazing Laetitia, the function is correct the only thing that you don't need here is the Parentheses.

File: week-1/Homework/extra/1-currency-conversion.js [hunk 2]
```diff
@@ -16,8 +19,13 @@ function convertToUSD() {}
   Find a way to add 1% to all currency conversions (think about the DRY principle)
 */
 
-function convertToBRL() {}
+function convertToBRL(priceInPound) {
+let priceInBRL = (priceInPound * 5.7);
+let fee = priceInBRL * 0.01;
+return priceInBRL + fee;
+}
```
Comment (Abdoulrazack95) [inline week-1/Homework/extra/1-currency-conversion.js:26]: well done here, one thing you don't need for the parentheses.

File: week-1/Homework/extra/2-piping.js [hunk 3]
```diff
@@ -16,27 +16,28 @@
...
+const badCode = (format(multiply(2,(add(startingValue, 10))))); //because it's difficult to type and read. 
 
+/* BETTER PRACTICE */
+let addOp = add(startingValue, 10);
+let multiplyOp = multiply(addOp, 2);
+let goodCode = format(multiplyOp);
+ 
 /* ======= TESTS - DO NOT MODIFY ===== */
 
 function test(test_name, expr) {
```
Comment (Abdoulrazack95) [inline week-1/Homework/extra/2-piping.js:43]: Great work, Perfect.

File: week-1/Homework/mandatory/1-syntax-errors.js [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
-    return "The remainder is %{remainder}"
+    return `The remainder is ${remainder}`
+   
 }
 
+console.log(getRemainder(10, 4));
+
 /* ======= TESTS - DO NOT MODIFY ===== */
 
 function test(test_name, expr) {
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/1-syntax-errors.js:28]: Good work Laetitia.

File: week-1/Homework/mandatory/1-syntax-errors.js [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+function introduceMe(name, age) {
+return "Hello, my name is " + name + " and I am " + age + " years old";
+
+}
+
+console.log(introduceMe("Laetitia", 30));
 
 function getRemainder(a, b) {
-    remainder = a %% b;
+    remainder = a % b;
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/1-syntax-errors.js:17]: You forgot the keyword variable here.

File: week-1/Homework/mandatory/2-logic-error.js [hunk 6]
```diff
@@ -1,18 +1,24 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function trimWord(word) {
-  return wordtrim();
+  
+  return word.trim();
 }
+console.log(trimWord("   Salut  "));
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/2-logic-error.js:7]: Hahah salut I like this one.

File: week-1/Homework/mandatory/2-logic-error.js [hunk 7]
```diff
@@ -1,18 +1,24 @@
...
 
 function getWordLength(word) {
-  return "word".length()
+  
+  return word.length;
 }
+console.log(getWordLength("coucou"));
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/2-logic-error.js:13]: haha coucou

File: week-1/Homework/mandatory/2-logic-error.js [hunk 8]
```diff
@@ -1,18 +1,24 @@
...
-  a * b * c;
-  return;
+  return a * b * c;
 }
 
+console.log(multiply(1,1,1));
+
 /* ======= TESTS - DO NOT MODIFY ===== */
 
 function test(test_name, expr) {
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/2-logic-error.js:24]: Amazing work here.

File: week-1/Homework/mandatory/3-function-output.js [hunk 9]
```diff
@@ -1,18 +1,29 @@
 // Add comments to explain what this function does. You're meant to use Google!
+//The Math.random() function returns a floating-point, pseudo-random number in the range 0 to less than 1 (inclusive of 0, but not 1).
+//The getNumber function returns the product of that random number and 10. 
 function getNumber() {
-    return Math.random() * 10;
+    
+        return Math.random() * 10;
 }
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/3-function-output.js:7]: Perfect here.

File: week-1/Homework/mandatory/3-function-output.js [hunk 10]
```diff
@@ -1,18 +1,29 @@
...
+
 function concatenate(firstWord, secondWord, thirdWord) {
     // Write the body of this function to concatenate three words together
     // Look at the test case below to understand what to expect in return 
+    return firstWord.concat(" " + secondWord + " " + thirdWord);
 }
 
+console.log(concatenate('code', 'your', 'future'));
+
 /* ======= TESTS - DO NOT MODIFY ===== */
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/3-function-output.js:27]: Well done Laetita

File: week-1/Homework/mandatory/4-tax.js [hunk 11]
```diff
@@ -5,7 +5,16 @@
   Sales tax is 20% of the price of the product
 */
 
-function calculateSalesTax() {}
+function calculateSalesTax(price) {
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/4-tax.js:8]: Here Laetitia you should have multiplied by 20%. Fix this part of the function please

File: week-1/Homework/mandatory/4-tax.js [hunk 12]
```diff
@@ -17,7 +26,14 @@ function calculateSalesTax() {}
...
 */
 
-function formatCurrency() {}
+function formatCurrency(price) {
+const currency = "£";
+let number = calculateSalesTax(price);
+return currency + number.toFixed(2);
+}
+
+console.log(formatCurrency(10));
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/4-tax.js:35]: Well done here, just fix the above function.

File: week-1/Homework/mandatory/5-magic-8-ball.js [hunk 13]
```diff
@@ -13,49 +13,90 @@ The question can be anything, but the answers are fixed,
...
+    return "negative";
+  }
+  // very Negative
+  if (randomNumberFloor === 15 || randomNumberFloor === 16 || randomNumberFloor === 17 || randomNumberFloor === 18 || randomNumberFloor === 19) {
+    return "very negative";
+  }
+}
+
+
+console.log(checkAnswer());
```
Comment (Abdoulrazack95) [inline week-1/Homework/mandatory/5-magic-8-ball.js:96]: In this exercise the way you did is wrong, read the instruction carefully.

File: week-1/Homework/extra/2-piping.js [hunk 14]
```diff
@@ -16,27 +16,28 @@
...
-function format() {
-
+function format(num) {
+return `${ "£" + num}`;
 }
 
-const startingValue = 2
-
+const startingValue = 2;
 // Why can this code be seen as bad practice? Comment your answer.
```
Comment (Abdoulrazack95) [inline week-1/Homework/extra/2-piping.js:32]: perfect

File: week-1/Homework/mandatory/2-logic-error.js (hunk 6 above)
Comment (supertata) [reply week-1/Homework/mandatory/2-logic-error.js:7]: haha

File: week-1/Homework/extra/1-currency-conversion.js (hunk 1 above)
Comment (supertata) [reply week-1/Homework/extra/1-currency-conversion.js:11]: Fixed :)

File: week-1/Homework/mandatory/4-tax.js (hunk 11 above)
Comment (supertata) [reply week-1/Homework/mandatory/4-tax.js:8]: 20% is also equals to a fifth I believe, that's why I put " / 5"

**Pull request closed without merging**